File: WorkBook_5/6.cpp
#include <iostream>
using namespace std;

double f(double x)
{
    return 0.25*x*x*x + x - 2;
}

double foo(double lp = 0.0, double rp = 2.0, double eps = 0.001)
{
    double center = (lp + rp)/2.0;
    if (f(lp) * f(rp) < 0)
    {
        if (rp - lp < eps)
        {
            return center;
        }
        if (f(lp) * f(center) < 0)
        {
            return foo(lp, center, eps);
        }
        else
        {
            return foo(center, rp,eps);
        }
    }
    else 
    {
        cout << "Корня на отрезке нет";
        return 0;
    }
}

int main()
{
    cout << foo();
    return 0;
}

File: sem_1/WorkBook_ 3/6.py
def foo(a, b, p):
    return a - b*p

a = int(input())
b = int(input())
left_point = 0
right_point = a
while left_point < right_point:
    center = (left_point + right_point + 1) // 2
    if foo(a, b, center) >= 0:
        left_point = center
    else:
        right_point = center - 1

chastnoe = left_point
ostat = a - b*chastnoe
print("Частное: ", chastnoe)
print("Остаток: ", ostat)
